cdap-app-fabric/src/main/java/io/cdap/cdap/security/impersonation/DefaultUGIProvider.java

```diff
@@ -172,4 +200,52 @@ private File localizeKeytab(Location keytabLocation) throws IOException {
 
     return localKeytabFile.toFile();
   }
+
+  private Map<String, String> getRuntimeImpersonationProperties(NamespacedEntityId programId) {
```

Comment:
The method contract is not clear to me. What's in the returned map? Better add javadoc.